compare.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"golang.org/x/perf/benchstat"
+)
+
+const BenchstatPrefix = "Benchmark"
+
+// Compare compares runs from the same platform
+func Compare(s []string) {
+	builders := make(map[string]*strings.Builder)
+	for _, path := range s {
+		info, err := os.Stat(path)
+		if err != nil {
+			panic(err)
+		}
+
+		if !info.Mode().IsDir() {
+			panic(fmt.Errorf("[compare] %s is not a valid path", path))
+		}
+
+		files, err := ioutil.ReadDir(path)
+		if err != nil {
+			panic(err)
+		}
+
+		for _, f := range files {
+			if f.IsDir() {
+				name := f.Name()
+				p := filepath.Join(path, name)
+				if b, ok := builders[name]; ok {
+					addResult(b, p)
+				} else {
+					var sb strings.Builder
+					builders[name] = &sb
+					addResult(&sb, p)
+				}
+			}
+		}
+	}
+
+	c := &benchstat.Collection{
+		Alpha:      0.05,
+		AddGeoMean: false,
+		DeltaTest:  benchstat.UTest,
+	}
+
+	for key, b := range builders {
+		if err := c.AddFile(key, strings.NewReader(b.String())); err != nil {
+			panic(err)
+		}
+	}
+
+	tables := c.Tables()
+	var buf bytes.Buffer
+	benchstat.FormatText(&buf, tables)
+	os.Stdout.Write(buf.Bytes())
+}
+
+func addResult(b *strings.Builder, folderPath string) {
+	tr := readTestResult(filepath.Join(folderPath, "result.json"))
+
+	// Latencies
+	const LatenciesPrefix = BenchstatPrefix + "Latencies"
+	writeStat(b, LatenciesPrefix, "Total", "1", tr.Latencies.Total.Milliseconds())
+	writeStat(b, LatenciesPrefix, "Mean", "1", tr.Latencies.Mean.Milliseconds())
+	writeStat(b, LatenciesPrefix, "50th", "1", tr.Latencies.P50.Milliseconds())
+	writeStat(b, LatenciesPrefix, "90th", "1", tr.Latencies.P90.Milliseconds())
+	writeStat(b, LatenciesPrefix, "95th", "1", tr.Latencies.P95.Milliseconds())
+	writeStat(b, LatenciesPrefix, "95th", "1", tr.Latencies.P95.Milliseconds())
+	writeStat(b, LatenciesPrefix, "99th", "1", tr.Latencies.P99.Milliseconds())
+	writeStat(b, LatenciesPrefix, "Max", "1", tr.Latencies.Max.Milliseconds())
+	writeStat(b, LatenciesPrefix, "Min", "1", tr.Latencies.Min.Milliseconds())
```

Comment:
I spent a little while with the `reflect` package to try to programatically do this by recursively traversing the TestResult struct to extract out the values we needed, but I got stuck with some nil pointers exceptions and gave up - might revist on the weekend if I get some time, I think I almost figured it out.

If you're interested, I started with an allowlist (which could be configured as cmd line flags), and then did lookups to match fields on the struct. In the case of things like `P99`, I just recursed into struct fields and did the lookup there (as `P99` belongs to the `Latencies` struct).